index.html

```diff
@@ -311,72 +330,163 @@ <h2>The <dfn>PerformanceObserver</dfn> interface</h2>
+              <code>"undefined"</code>:
+                <ol>
+                  <li>If <var>options</var>'s <a>entryTypes</a> member is
+                  present, then set <var>observer</var>'s <a>observer type</a>
+                  to <code>"multiple"</code>.</li>
+                  <li>If <var>options</var>'s <a>type</a> member is present,
+                  then set <var>observer</var>'s <a>observer type</a> to
+                  <code>"single"</code>.</li>
+                </ol>
               </li>
-              <li>Append <var>entries</var> to the <a data-cite=
-              "WHATWG-DOM#context-object">context object</a>'s <a>observer
-              buffer</a>.
+              <li>If <var>observer</var>'s <a>observer type</a> is
+              <code>"single"</code> and <var>options</var>'s
+              <a>entryTypes</a> member is present, then
+              <a data-cite="!WEBIDL#dfn-throw">throw</a> an
+              <a data-cite="!WEBIDL#invalidmodificationerror">
+              <code>InvalidModificationError</code></a>.
+              </li>
+              <li>If <var>observer</var>'s <a>observer type</a> is
+              <code>"multiple"</code> and <var>options</var>'s <a>type</a>
+              member is present, then <a data-cite="!WEBIDL#dfn-throw">throw</a>
+              an <a data-cite="!WEBIDL#invalidmodificationerror">
+              <code>InvalidModificationError</code></a>.
               </li>
             </ol>
           </li>
+          <li>If <var>observer</var>'s <a>observer type</a> is
+          <code>"multiple"</code>, run the following steps:
+          <ol>
+            <li>Let <var>entry types</var> be <var>options</var>'s
+            <a>entryTypes</a> sequence.
+            </li>
+            <li>Remove all types from <var>entry types</var> that are not
+            contained in <a>supported entry types</a>. The user agent SHOULD
+            notify developers if <var>entry types</var> is modified. For
+            example, a console warning listing removed types might be
+            appropriate.</li>
+            <li>If the resulting <var>entry types</var> sequence is an empty
+            sequence, abort these steps. The user agent SHOULD notify developers
+            when the steps are aborted to notify that registration has been
+            aborted. For example, a console warning might be appropriate.</li>
+            <li>If the <a>list of registered performance observer objects</a> of
+            <a data-cite="HTML/webappapis.html#concept-relevant-global">relevant
+            global object</a> contains a <a>registered performance observer</a>
+            whose <a>observer</a> is the <a data-cite=
+            "WHATWG-DOM#context-object">context object</a>, replace its
+            <a>options list</a> with a list containing <var>options</var> as its
+            only item.</li>
+            <li>Otherwise, append a new <a>registered performance observer</a>
```

Comment:
create and append?

Reply:
Hm sure, but then probably remove the 'new' since it would be redundant?